### DISCUSSION TITLE
Add WebAssembly support

### DISCUSSION THREAD

**Discussion by Razican** [boa-dev/boa#1534] [Ideas]
WASM is now in everywhere, and a JavaScript engine should include support for it, I think. We could take [this][article] for ideas.

[article]: https://medium.com/@richardanaya/lets-write-a-web-assembly-interpreter-part-1-287298201d75

**Comment by jasonwilliams** [comment]
@Razican what needs to change here that we're not doing already?

**Comment by Razican** [comment]
We don't have any support for executing webassembly binaries inside of Boa. It would be nice to be able to run Boa in Boa, for example.

**Comment by Razican** [comment]
We could potentially use https://crates.io/crates/wasmtime for this, but we will need a module system first.

**Comment by jedel1043** [comment]
Bumping and pinning this since now that we have a module system, this should be reachable :)
